src/arizona_template_compiler.erl

```diff
@@ -0,0 +1,646 @@
+-module(arizona_template_compiler).
+-moduledoc false.
+
+%% --------------------------------------------------------------------
+%% API function exports
+%% --------------------------------------------------------------------
+
+-export([compile/3]).
+
+%
+
+-ignore_xref([compile/3]).
+
+%% --------------------------------------------------------------------
+%% Types (and their exports)
+%% --------------------------------------------------------------------
+
+-record(state, {
+    module :: module(),
+    function :: atom(),
+    macros :: macros(),
+    attributes :: [arizona_template_parser:attribute()],
+    index :: non_neg_integer(),
+    path :: [non_neg_integer()]
+}).
+
+-type macros() :: #{atom() := arizona_html:safe_type()}.
```

Comment:
Values of macros can be any term. They only replace variables and are not the final render value.
```suggestion
-type macros() :: #{atom() := term()}.
```

Reply:
[ddb06fa](https://github.com/arizona-framework/arizona/pull/164/commits/ddb06faa953bdba094e2c39b6d793baa51f6ac51)